nodes/src/nodes/eval_cobalt/evaluators/relevance.py

```
@@ -0,0 +1,154 @@
+    if not output:
+        return {'score': 0.0, 'passed': False, 'reasoning': 'Output is empty but expected content was provided.'}
+
+    output_tokens = _tokenize(output)
+    expected_tokens = _tokenize(expected)
+
+    keyword_score = _compute_keyword_overlap(output_tokens, expected_tokens)
+    length_score = _compute_length_ratio(output, expected)
+
+    total_weight = keyword_weight + length_weight
+    if total_weight <= 0:
+        raise ValueError('keyword_weight + length_weight must be > 0')
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)